### PR TITLE
ATO-981 set is new account in userinfo

### PR DESCRIPTION
**Pull request by Ryan-Andrews99**
## What
- We're now setting the value of `isNewAccount` in the Auth userInfo lambda. This is a pre-requisite to splitting this value from the shared session store as currently Auth rely on Orchestration to update this in `AuthenticationCallBack` handler and `IPVCallBack` handler. We're setting it here as Auth need to start independently setting this value, and this point is closest to the end of the handoff between Auth and Orch, and closest to when it would be set by Orch.
- Soon we will co-set this value in the new Auth session store, before eventually migrating to the auth only store.

## How to review

For example:
- Code review commit-by-commit 
- Deployed to sandpit and ran through a journey, confirmed that the session is updated in the userinfo handler

## Checklist

- [x] Impact on orch and auth mutual dependencies has been checked.

- [ ] A UCD review has been performed.

### CODE REVIEW

File: auth-external-api/src/test/java/uk/gov/di/authentication/external/lambda/UserInfoHandlerTest.java [hunk 1]
```diff
@@ -77,18 +85,22 @@ void shouldReturn200WithUserInfoForValidRequestAndSetTokenStoreUsed()
...
 
         APIGatewayProxyResponseEvent response = userInfoHandler.userInfoRequestHandler(request);
+        Session updatedSession = testSession.setNewAccount(Session.AccountState.EXISTING);
 
         assertEquals(200, response.getStatusCode());
         assertTrue(
                 response.getBody()
                         .contains(String.format("\"sub\":\"%s\"", TEST_SUBJECT.getValue())));
 
+        verify(sessionService).getSessionFromRequestHeaders(request.getHeaders());
```
Comment (BeckaL) [inline auth-external-api/src/test/java/uk/gov/di/authentication/external/lambda/UserInfoHandlerTest.java:103]: Do we need this assertion? This feels to me a bit too much like testing the internals of a function. We set up what this should return on line 92, that should be enough.

File: auth-external-api/src/main/java/uk/gov/di/authentication/external/lambda/UserInfoHandler.java [hunk 2]
```diff
@@ -186,6 +186,8 @@ public APIGatewayProxyResponseEvent userInfoRequestHandler(APIGatewayProxyReques
                     accessToken.getValue());
         }
 
+        sessionService.save(userSession.setNewAccount(Session.AccountState.EXISTING));
+        LOG.info("Set account state to EXISTING");
```
Comment (BeckaL) [inline auth-external-api/src/main/java/uk/gov/di/authentication/external/lambda/UserInfoHandler.java:190]: Is this logging useful?

File: auth-external-api/src/main/java/uk/gov/di/authentication/external/lambda/UserInfoHandler.java [hunk 3]
```diff
@@ -186,6 +186,8 @@ public APIGatewayProxyResponseEvent userInfoRequestHandler(APIGatewayProxyReques
                     accessToken.getValue());
         }
 
+        sessionService.save(userSession.setNewAccount(Session.AccountState.EXISTING));
```
Comment (BeckaL) [inline auth-external-api/src/main/java/uk/gov/di/authentication/external/lambda/UserInfoHandler.java:189]: Is the logic here that as long as the request is successful, any account (including new accounts) are marked as existing at this point?

Could you do some combo (maybe both :-)) of:
* writing a bit more about why this behaviour is useful in the commit message
* call it out as a specific scenario in a test (updates a new account to existing when user info handler request processed correctly?)

**Review by BeckaL — COMMENTED** [review]
Nice! Thanks for an easily digestible review. A couple of comments / questions / suggestions below

File: auth-external-api/src/test/java/uk/gov/di/authentication/external/lambda/UserInfoHandlerTest.java (hunk 1 above)
Comment (Ryan-Andrews99) [reply auth-external-api/src/test/java/uk/gov/di/authentication/external/lambda/UserInfoHandlerTest.java]: Yeah I see what you mean, feels a bit unnecessary, have removed!

File: auth-external-api/src/main/java/uk/gov/di/authentication/external/lambda/UserInfoHandler.java (hunk 2 above)
Comment (Ryan-Andrews99) [reply auth-external-api/src/main/java/uk/gov/di/authentication/external/lambda/UserInfoHandler.java]: It was useful for testing, doesn't particularly tell you anything useful outside of "the code did its job" which is expected. Will remove 🙂

File: auth-external-api/src/main/java/uk/gov/di/authentication/external/lambda/UserInfoHandler.java (hunk 3 above)
Comment (Ryan-Andrews99) [reply auth-external-api/src/main/java/uk/gov/di/authentication/external/lambda/UserInfoHandler.java:189]: I have added a separate test case for this and tried to elaborate a bit more in the commit message. Let me know if commit message is too context-y and not explain-y enough 😅

**Review by BeckaL — APPROVED** [review]
Lovely stuff!

**Review by billrdunn — DISMISSED** [review]
A beautiful PR  👏

**Review by BeckaL — APPROVED** [review]

**Pull request merged**